huaweicloud/resource_huaweicloud_evs_volume.go

```diff
@@ -268,6 +253,41 @@ func resourceEvsVolumeV3Read(d *schema.ResourceData, meta interface{}) error {
 	if err := d.Set("attachment", attachments); err != nil {
 		return fmtp.Errorf("[DEBUG] Error saving attachment to state for HuaweiCloud evs storage (%s): %s", d.Id(), err)
 	}
+	return nil
+}
+
+func resourceEvsVolumeV3Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	blockStorageClient, err := config.BlockStorageV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmtp.Errorf("Error creating HuaweiCloud EVS storage client: %s", err)
+	}
+
+	resp, err := volumes.Get(blockStorageClient, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "volume")
+	}
+
+	logp.Printf("[DEBUG] Retrieved volume %s: %+v", d.Id(), resp)
+	mErr := multierror.Append(
+		d.Set("name", resp.Name),
+		d.Set("size", resp.Size),
+		d.Set("description", resp.Description),
+		d.Set("availability_zone", resp.AvailabilityZone),
+		d.Set("snapshot_id", resp.SnapshotID),
+		d.Set("volume_type", resp.VolumeType),
+		d.Set("enterprise_project_id", resp.EnterpriseProjectID),
+		d.Set("region", GetRegion(d, config)),
+		d.Set("wwn", resp.WWN),
+		d.Set("multiattach", resp.Multiattach),
+		setEvsVolumeDeviceType(d, resp),
+		setEvsVolumeImageId(d, resp),
+		setEvsVolumeTags(d, resp),
+		setEvsVolumeAttachment(d, resp),
+	)
+	if err := mErr.ErrorOrNil(); err != nil {
```

Comment:
`err` has been declared above and can be used directly here.

Reply:
done